samples/query_log/plugin.cc

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_query_log]
+#include <boost/url/parse.hpp>
+#include <boost/url/url.hpp>
+
+#include "proxy_wasm_intrinsics.h"
+
+class MyHttpContext : public Context {
+ public:
+  explicit MyHttpContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders(uint32_t headers,
+                                       bool end_of_stream) override {
+    auto path = getRequestHeader(":path");
```

Comment:
tiny nit: I assume we're following the Google c++ style guide, and per its guidance on `auto`, I think it'd be better to spell out the type; especially since this is a proxy_wasm type that users are likely to use.

https://google.github.io/styleguide/cppguide.html#Type_deduction

Reply:
Fair point, replaced here and below for boost types.